rotkehlchen/db/snapshots.py

```diff
@@ -0,0 +1,108 @@
+import csv
+import tempfile
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+from zipfile import ZipFile
+
+from rotkehlchen.accounting.export.csv import CSVWriteError
+from rotkehlchen.accounting.structures.balance import BalanceType
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.db.utils import DBAssetBalance, LocationData
+from rotkehlchen.types import Location, Timestamp
+
+
+class Snapshot:
+    def __init__(self, db_handler: DBHandler) -> None:
+        self.db_handler = db_handler
+        self.tmp_dir = tempfile.TemporaryDirectory()
+        self.tmp_dir_path = Path(self.tmp_dir.name)
+        self.snapshot_archive_path = self.tmp_dir_path / 'snapshot.zip'
+        self.timed_balances_csv_path = self.tmp_dir_path / 'timed_balances.csv'
+        self.timed_location_data_csv_path = self.tmp_dir_path / 'timed_location_data.csv'
```

Comment:
```suggestion
```

Remove those. They should not be initialized here. This is a temporal object. The temporary directory and files should only be created when/if needed, not at initiailzation.